sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -289,14 +289,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
   }
 
   def namedArgumentsNotEnabledError(functionName: String, argumentName: String): Throwable = {
-    new AnalysisException(
+    new SparkUnsupportedOperationException(
       errorClass = "NAMED_PARAMETER_SUPPORT_DISABLED",
       messageParameters = Map(
         "functionName" -> toSQLId(functionName),
         "argument" -> toSQLId(argumentName))
     )
   }
 
+  def collationDisabledError(): Throwable = {
```

Comment:
In the same spirit as my previous comment, I would rename this to `collationNotEnabledError`.